Выполнено основное задание к 3-му спринту.

AnastasiyaIlyina opened the pull request.

AnastasiyaIlyina committed:
Выполнено основное задание к 3-му спринту.

egorvas commented on src/test/java/com/example/CourierAuthorizationTest.java:
```diff
@@ -0,0 +1,89 @@
+    }
+
+    @Test
+    public void LoginCourierResponseOk() {
```
test это функция, поэтому тест должен начинаться с маленькой буквы

AnastasiyaIlyina replied:
++

egorvas commented on src/test/java/com/example/utils/CourierAuthenticator.java:
```diff
@@ -0,0 +1,23 @@
+public class CourierAuthenticator {
+
+    public Response loginCourier(String login, String password){
+        String preprocessedLogin = login == null ? "null" : "\"" + login + "\"";
```
Это что еще за жесть? :)
Нельзя было попроще сделать это?

AnastasiyaIlyina replied:
Если бы воспользовалась методом о котором вы говорили https://github.com/AnastasiyaIlyina/Sprint_3/pull/1#discussion_r769008540
то удалось бы написать проще.

egorvas commented on src/test/java/com/example/utils/CourierAuthenticator.java:
```diff
@@ -0,0 +1,23 @@
+                .and()
+                .body(loginRequestBody)
+                .when()
+                .post("https://qa-scooter.praktikum-services.ru/api/v1/courier/login");
```
baseURI("https://qa-scooter.praktikum-services.ru" ) лучше выносить в одно общее место для всех тестов. На практике твои тесты будут запускаться на разных энвах: дев, прод, тест и тп и везде будут разные url. По этому чтобы в каждом тесте не менять baseURI лучше это делать один раз для всех тестов.

AnastasiyaIlyina replied:
++

egorvas commented on src/test/java/com/example/utils/CourierOrderAcceptor.java:
```diff
@@ -0,0 +1,17 @@
+        return given()
+                .header("Content-type", "application/json")
+                .when()
+                .put("https://qa-scooter.praktikum-services.ru/api/v1/orders/accept/"+ trackId.toString() + "?courierId=" + courierId.toString());
```
courierId это env и его нужно передавать через .queryParam()

AnastasiyaIlyina replied:
++

egorvas commented on src/test/java/com/example/utils/CourierRegistrar.java:
```diff
@@ -0,0 +1,48 @@
+        String preprocessedFirstName = firstName == null ? "null" : "\"" + firstName + "\"";
+
+
+        String registerRequestBody = "{\"login\":" + preprocessedLogin + ","
```
Я ярый противник формирования JSON body вот через такие строки состоящие из скобочек, обратных слешей и тп. Лучше всего генерировать боди через сериализацию с помощью модулей вроде jackson или gson.  Или есть более простой способ, можно генерировать JSON через JAVA класс Map или с помощью стороннего модуля JSON.
Я не могу считать это оишбкой, но я крайне рекомендую избавится от всех таких строк.

AnastasiyaIlyina replied:
Воспользуюсь тем, что это не ошибка., так как в теории использовался такой подход.
Обязательно изучу этот подход.
В обратную связь добавлю коменнтарий о плохом подходе.

egorvas commented on src/test/java/com/example/OrderListTest.java:
```diff
@@ -0,0 +1,74 @@
+    }
+
+    @Test
+    public void LoginCourierResponseOk() {
```
Я не понимаю почему этот тест так называется, а еще я не понимаю почему он падает у тебя в отчете

AnastasiyaIlyina replied:
++

egorvas commented on src/test/java/com/example/CourierAuthorizationTest.java:
```diff
@@ -0,0 +1,89 @@
+        Response response = courierAuthenticator.loginCourier(randomCourierLoginPass.get(0), null);
+        response.then()
+                .assertThat()
+                .body("message", equalTo("Недостаточно данных для входа"))
```
У тебя тест падает с очень некрасивой ошибкой 
![image](https://user-images.githubusercontent.com/7426788/146073356-20d42827-7b8c-4fe6-956f-e932d27e2932.png)
Это происходит потому что на сервис виснет при таком запросе и не отдает тело запроса, но статус код отдает. Поэтому я всегда рекомендую первым делом проверять статус код, а потом всегда все остальное. Если первым проверять статус код, то тест будет падать с красивой ошибкой статус кода.

AnastasiyaIlyina replied:
++

AnastasiyaIlyina committed:
Внесены исправления:
1. Изменены названия тестов
2. Вынесен BASE_URL
3. Добавлен .queryParam()
4. Исправлено падение теста регистрации курьера
5. Исправлен отчет Allure
6. Изменен порядок проверок статуса и тела ответа
